### PR TITLE
Add filter dropdown and change task card style in mine page

### LINKED ISSUE #1262: [Fix]: mine page
### Issue Description
Mine page does not have filter dropdown component and  task card looks different than the tasks page

### Expected Behavior
User should be able to see filter dropdown component and task card same as tasks page
image

### Current Behavior
User can not see filter dropdown component and task card looks different than the tasks page

### Screenshots
image

### Reproducibility
 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. Go to status site
2. Go to mine page

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by pankajjs**
Date:  19 Sep 2024

Developer Name: Pankaj Sha

---

## Issue Ticket Number

- Closes #1262 

## Description

This pr adds a filter dropdown component, changes the style of task card and implements filter logic. 

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Working video</summary>

https://github.com/user-attachments/assets/56dadf8b-a767-4b83-aa7a-f90ff74e2f83

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-09-26 at 6 20 11 PM

</details>
<details>
<summary>Screenshot 2</summary>
Screenshot 2024-09-27 at 4 55 42 PM

</details>

### CODE REVIEW

**Comment by vinit717** [comment]
Did you design doc got approved?

**Comment by Ayushsanjdev** [comment]
1. PR name doesn't make any sense.
2. Screen recording doesn't play.
@pankajjs

**Comment by pankajjs** [comment]
> Did you design doc got approved?

Not approved till now. But you asked me to do some changes. I did.

**Comment by pankajjs** [comment]
> anges. I did.

**Review by iamitprakash — CHANGES_REQUESTED** [review]
don't we need FF here

**Comment by vinit717** [comment]
Where is TDD PR?
as mentioned in the design doc you are following TDD right

**Comment by pankajjs** [comment]
> Where is TDD PR? as mentioned in the design doc you are following TDD right

I have followed TDD. However I was not aware that I have to raise pr for TDD first.  This pr contains every changes.

**Comment by pankajjs** [comment]
> don't we need FF here

Added FF.

File: __tests__/Unit/utils/getFilteredTasks.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        title: 'Testing and Determinsitic State',
+        purpose: 'string',
+        percentCompleted: 0,
+        endsOn: 1618790400,
+        status: 'progress',
+        featureUrl: 'progress',
+        type: 'feature',
+        createdBy: 'shmbajaj',
+        priority: 'TBD',
+    };
```
Comment (vinit717) [inline __tests__/Unit/utils/getFilteredTasks.test.ts:30]: Can you add this on fixture

Comment (pankajjs) [reply __tests__/Unit/utils/getFilteredTasks.test.ts]: Resolved.

**Review by iamitprakash — APPROVED** [review]

**Comment by vinit717** [comment]
@pankajjs are we not using the same component for the task tab 
as the UI and functionality are the same

**Comment by pankajjs** [comment]
> @pankajjs are we not using the same component for the task tab as the UI and functionality are the same
1. Tasks page uses this two components with its own filter logic as it receives data in chunks.
Screenshot 2024-09-22 at 11 33 18 PM
Screenshot 2024-09-22 at 11 33 32 PM

2. In Mine page, we receive data at once. Therefore, we are using these two components with its own filter logic.

**Comment by vinit717** [comment]
> > @pankajjs are we not using the same component for the task tab as the UI and functionality are the same
> 
> 1. Tasks page uses this two components with its own filter logic inside a component named TasksContent as it receives data in chunks while scrolling.
> 
> Screenshot 2024-09-22 at 11 33 18 PM Screenshot 2024-09-22 at 11 33 32 PM
> 2. In Mine page, we receive data at once. Therefore, we are using these two components with its own filter logic.
> 3. I have made the functionality looks same even the logic is different because of its data fetching technique.

But self API should be paginate then as by this we don't need to create another component for this we can reuse it
can you confirm this with @ankushdharkar

**Comment by pankajjs** [comment]
@vinit717 Even If we make self API paginated, we need another component to make the filter and search should work as same as Tasks page. Here we are just changing the user's tasks state on filter and search change.

File: src/pages/mine/index.tsx [hunk 2]
```diff
@@ -1,14 +1,82 @@
...
+
+    const filterTabHandler = (selectedTab: Tab) => {
+        searchTasks(getInputValueFromTaskField(selectedTab, title));
+    };
+
+    useEffect(() => {
+        setFilteredTasks(tasks);
+    }, [tasks]);
+
+    if (isLoading) return <p>Loading...</p>;
```
Comment (vinit717) [inline src/pages/mine/index.tsx:56]: Can we show the shimmer component when loading

**Comment by vinit717** [comment]
> @vinit717 Even If we make self API paginated, we need another component to make the filter and search should work as same as Tasks page because the TaskContent is tightly coupled with the hook which returns the tasks list.

But we need to make the component dumb so it get the data and show accordingly 
for now we can merge your PR and If you want,  make the backend API paginated and change the frontend logic accordingly

File: src/utils/getInputValueFromTaskField.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Tab } from '@/interfaces/task.type';
+import { getQueryParamTab, getQueryParamTitle } from './taskQueryParams';
+
+export default function getInputValueFromTaskField(tab: Tab, title: string) {
+    let inputValue = '';
+
+    if (tab) inputValue += `${getQueryParamTab(tab)} `;
```
Comment (vinit717) [inline src/utils/getInputValueFromTaskField.ts:7]: This line is uncover in test

File: src/pages/mine/index.tsx
Comment (vinit717) [inline src/pages/mine/index.tsx:1]: Improve the test coverage for this file

**Comment by pankajjs** [comment]
> > @vinit717 Even If we make self API paginated, we need another component to make the filter and search should work as same as Tasks page because the TaskContent is tightly coupled with the hook which returns the tasks list.
> 
> But we need to make the component dumb so it get the data and show accordingly for now we can merge your PR and If you want, make the backend API paginated and change the frontend logic accordingly

We can merge this Pr. For the backend, I will pick it another task.

File: __tests__/Unit/utils/getFilteredTasks.test.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
```
Comment (vinit717) [inline __tests__/Unit/utils/getFilteredTasks.test.ts:35]: can you move this to fixture

File: src/pages/mine/index.tsx
Comment (vinit717) [inline src/pages/mine/index.tsx:1]: can you increase the test coverage for this file

Comment (pankajjs) [reply src/pages/mine/index.tsx:1]: Before changes
Screenshot 2024-09-26 at 6 20 38 PM

After changes
Screenshot 2024-09-27 at 4 55 42 PM

**Review by vinit717 — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**